build/git.go

```diff
@@ -93,23 +99,40 @@ func getGitAttributes(ctx context.Context, contextPath string, dockerfilePath st
 		}
 	}
 
-	if setGitLabels {
-		if root, err := gitc.RootDir(); err != nil {
-			return res, errors.Wrap(err, "failed to get git root dir")
-		} else if root != "" {
-			if dockerfilePath == "" {
-				dockerfilePath = filepath.Join(wd, "Dockerfile")
+	if setGitLabels && root != "" {
+		if dockerfilePath == "" {
+			dockerfilePath = filepath.Join(wd, "Dockerfile")
+		}
+		if !filepath.IsAbs(dockerfilePath) {
+			cwd, _ := os.Getwd()
+			dockerfilePath = filepath.Join(cwd, dockerfilePath)
+		}
+		dockerfilePath, _ = filepath.Rel(root, dockerfilePath)
+		if !strings.HasPrefix(dockerfilePath, "..") {
```

Comment:
`..` + `filepath.Separator` ?

Reply:
It was added initially in https://github.com/docker/buildx/pull/1297 but instead we should just check if `.Rel` returns an error

Reply:
No, `Rel` can return `../` paths without error and they should not be allowed.

Reply:
:man_facepalming: